.github/workflows/todo_app_ci_pipeline.yml

```diff
@@ -11,11 +11,35 @@ on:
 
 jobs:
   build:
-    name: Build and test
+    name: Build & Test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - run: docker build --target test --tag my-test-image .
     - run: docker run --env-file .env.test my-test-image todo_app/tests
     - run: docker run -e trello_board_id=${{secrets.TRELLO_BOARD_ID}} -e trello_key=${{secrets.TRELLO_KEY}} -e trello_token=${{secrets.TRELLO_TOKEN}}  my-test-image todo_app/e2e_tests
 
+  publish:
+    name: Push Image to DockerHub
+    needs: build
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/exercise-8' && github.event_name == 'push'
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build and push to Heroku
+      id: heroku
+      uses: jctaveras/heroku-deploy@v1.0.0 
+      with:
+        email: ${{ secrets.HEROKU_EMAIL }} 
+        api_key: ${{ secrets.HEROKU_API_KEY }} 
+        app_name: ${{ secrets.HEROKU_APP_NAME }} 
+        dockerfile: './' # set the path to the folder where the Dockerfile is located
+        options: '--target production --tag jaisalpatel836/todo_app:latest --tag jaisalpatel836/todo_app:"$GITHUB_SHA"' # current version of action doesn't support this - need to upgrade - only works since prod is last base to build from
+    - uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - run: docker build --target production --tag jaisalpatel836/todo_app:latest --tag jaisalpatel836/todo_app:"$GITHUB_SHA" .
+    - run: docker push jaisalpatel836/todo_app --all-tags
```

Comment:
I can't see this image in DockerHub but I can see your action pushing to it, so I'm assuming it's just a private image!

Reply:
it is indeed a private image
<img width="944" alt="Screenshot 2022-03-29 at 21 47 36" src="https://user-images.githubusercontent.com/18398852/160704299-44fd4ea7-cecf-44f0-8bfb-b0b47d1e7bdd.png">